### PR TITLE
Mobile: Fixes #3533: Add marker to Geolocation URL

### LINKED ISSUE #3533: note geolocation set to 0. 0. when shared with OsmAnd~
## Environment

Joplin version: 1.0.333
Platform: Android
OS specifics: 9.1.0

## Steps to reproduce

1. create a note
2. check properties, tap View on map
3. choose OsmAnd~
4. map shown in the middle of the ocean.  check details: 0.0000S 0.0000W

Note: if choose Chrome in Step 3, then location seems correct.

### PR DESCRIPTION
**Pull request by reinhart1010**
In this PR, I modified the geolocation (OpenStreetMap) URL at `ReactNativeClient/lib/models/Note.js` to:

1. Display a marker on the OpenStreetMap preview. This might sound trivial, but this could be more useful to exactly identify the location where the note was added.
2. Fixes #3533, where the URL was unable to be opened properly with OSMAnd, an external Android app which also reads the `https://www.openstreetmap.org` URI/URL through intent filters.

image

When a mobile app user opens the "View on Map" on note properties, it will now open in a link similar to
```
https://www.openstreetmap.org/?mlat=-6.19552&mlon=106.82281&zoom=20
```
instead of
```
https://www.openstreetmap.org/?lat=-6.19552&lon=106.82281&zoom=20
```

### CODE REVIEW

**Comment by reinhart1010** [comment]
I was unable to test this on Android since I have setup issues with JDK, Gradle (this project's Gradle configuration has been old enough), and React Native, but feel free to suggest any changes before merging to `master`.

File: BUILD.md [hunk 1]
```diff
@@ -93,4 +93,4 @@ The Markdown renderer is located under ReactNativeClient/lib/joplin-renderer. Wh
 
 # Troubleshooting
 
-Please read for the [Build Troubleshooting Document](https://github.com/laurent22/joplin/blob/master/readme/build_troubleshooting.md) for various tips on how to get the build working.
\ No newline at end of file
+Please read for the [Build Troubleshooting Document](https://github.com/laurent22/joplin/blob/master/readme/build_troubleshooting.md) for various tips on how to get the build working.
```
Comment (laurent22) [inline BUILD.md]: Thank you for the pull request, it makes sense. Please could you just remove this change from the pull request as it's unrelated.

**Review by laurent22 — CHANGES_REQUESTED** [review]

File: BUILD.md (hunk 1 above)
Comment (reinhart1010) [reply BUILD.md]: Apologize for that. I guess that it was due to a merge conflict between the current codebase and the ones I have for PR #1922. I am trying to fix this.

**Comment by laurent22** [comment]
Many thanks for the update!

**Comment by reinhart1010** [comment]
You're welcome. Can you add `#hacktoberfest-accepted` label too (similar to #3930)?

**Comment by laurent22** [comment]
Sure, that's done.

**Pull request merged**